.github/workflows/ci.yml

```diff
@@ -317,3 +320,82 @@ jobs:
         run: |
           echo "Postman tests failed with exit code ${{ steps.postman_tests.outputs.exit_code }}"
           exit ${{ steps.postman_tests.outputs.exit_code }}
+
+  test-e2e:
+    name: test-e2e
+    runs-on: ubuntu-latest
+    needs: changes
+    if: ${{ needs.changes.outputs.server == 'true' || needs.changes.outputs.client == 'true' || needs.changes.outputs.test_e2e == 'true' }}
+    env:
+      DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    steps:
+      - name: Explain skip
+        if: ${{ needs.changes.outputs.server != 'true' && needs.changes.outputs.client != 'true' && needs.changes.outputs.test_e2e != 'true' }}
+        run: echo "No App/Client, App/Server, or Test/e2e changes detected; skipping E2E tests."
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup .NET (from global.json)
+        uses: actions/setup-dotnet@v4
+        with:
+          global-json-file: App/Server/global.json
+          cache: true
+          cache-dependency-path: "**/packages.lock.json"
+
+      - name: Verify .NET version
+        run: dotnet --info
+
+      - name: Test Server (E2E)
+        id: e2e_tests
+        run: |
+          set +e  # Don't exit on error
+          ./build.sh test-server-e2e
+          TEST_EXIT_CODE=$?
+          echo "exit_code=$TEST_EXIT_CODE" >> $GITHUB_OUTPUT
+          exit 0  # Always succeed so we can upload artifacts and comment
+
+      - name: Upload E2E Test Reports
+        uses: actions/upload-artifact@v4
+        if: always()  # Upload even if tests failed
+        with:
+          name: e2e-test-reports
+          path: |
+            TestResults/*.trx
+            TestResults/traces/*.zip
```

Comment:
This might clash with existing trx reports being lumped into the same directory. Can you update it so that for the Test/e2e tests it outputs to Reports/e2e instead? Leave the existing xunit report paths untouched, we'll refactor that in a separate PR.